### PR TITLE
Import redeclared as function name inspection (fixes #1198)

### LINKED ISSUE #1198: [1.0] Symbol redeclare not detected
```go
package main

import "fmt"

func fmt() {
}

func main() {
	fmt.Println("Hello, playground")
}
```

go show those errors:

```
prog.go:5: fmt redeclared in this block
	previous declaration at prog.go:3
prog.go:9: fmt.Println undefined (type func() has no field or method Println)
```

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

**Comment by dlsniper** [comment]
Sorry for the empty lines diffs, I can try to remove them if needed. This is ready for review. As a general question, should I focus on doing the inspections for now and then start with the quickfixes or should I always do them in pairs. Thank you.

File: src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public class GoRedeclareImportAsFunctionInspection extends GoInspectionBase {
+  @Override
+  protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    if (GoTestFinder.getTestTargetPackage(file) != null) return;
+
+    List<GoFunctionDeclaration> functions = file.getFunctions();
+
+    for (GoFunctionDeclaration function : functions) {
+      for (GoImportSpec importSpec : file.getImports()) {
+        if (function.getIdentifier().textMatches(importSpec.getLocalPackageName())) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java]: This breaks stubs, use function.getName instead

**Comment by zolotov** [comment]
@dlsniper as you wish. I prefer making inspection with quick fixes in pairs

**Comment by dlsniper** [comment]
Done, I've updated the PR with the feedback and the a rename quick fix.

**Comment by zolotov** [comment]
Thanks, will merge this on the weekend, sorry for delay

File: src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    if (GoTestFinder.getTestTargetPackage(file) != null) return;
+
+    List<GoFunctionDeclaration> functions = file.getFunctions();
+    String funcName;
+
+    for (GoFunctionDeclaration function : functions) {
+      funcName = function.getName();
+      if (funcName == null) continue;
+      for (GoImportSpec importSpec : file.getImports()) {
```
Comment (sjamesr) [inline src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java]: should this loop be replaced by

```java
if (file.getImportMap().containsKey(funcName)) {
  problemsHolder.registerProblem(...);
```

?

Comment (zolotov) [inline src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java]: @sjamesr indeed, thank you.

File: src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class GoRedeclareImportAsFunctionInspection extends GoInspectionBase {
+  @Override
+  protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    if (GoTestFinder.getTestTargetPackage(file) != null) return;
```
Comment (zolotov) [inline src/com/goide/inspections/GoRedeclareImportAsFunctionInspection.java]: Didn't get why is that. It's also untested. Removed.

**Comment by dlsniper** [comment]
@zolotov thanks for merging. I've been really busy this weekend I couldn't act on the feedback. @sjamesr thanks for the feedback as well.

**Pull request merged**